src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingInfo.cs

```diff
@@ -134,6 +138,70 @@ public static BindingInfo GetBindingInfo(IEnumerable<object> attributes)
             return isBindingInfoPresent ? bindingInfo : null;
         }
 
+        /// <summary>
+        /// Constructs a new instance of <see cref="BindingInfo"/> from the given <paramref name="attributes"/> and <paramref name="modelMetadata"/>.
+        /// <para>
+        /// <see cref="GetBindingInfo(IEnumerable{object})"/> does not account for binding information configured using
+        /// <see cref="ModelMetadata"/>.
+        /// </para>
+        /// </summary>
+        /// <param name="attributes">A collection of attributes which are used to construct <see cref="BindingInfo"/>.</param>
+        /// <param name="modelMetadata">The <see cref="ModelMetadata"/>.</param>
+        /// <returns>A new instance of <see cref="BindingInfo"/>.</returns>
+        public static BindingInfo GetBindingInfo(IEnumerable<object> attributes, ModelMetadata modelMetadata)
+        {
+            if (attributes == null)
+            {
+                throw new ArgumentNullException(nameof(attributes));
+            }
+
+            if (modelMetadata == null)
+            {
+                throw new ArgumentNullException(nameof(modelMetadata));
+            }
+
+            var bindingInfo = GetBindingInfo(attributes);
+            if (bindingInfo != null)
+            {
+                return bindingInfo;
+            }
```

Comment:
Is this condition right? What if the MM defines something that the binding info doesn't?